merkle-tree/concurrent/src/lib.rs

```diff
@@ -801,11 +826,14 @@ where
             node = compute_parent_node::<H>(&node, sibling, leaf_index, j)?;
         }
 
+        self.sequence_number = self
+            .sequence_number
+            .checked_add(1)
+            .ok_or(ConcurrentMerkleTreeError::IntegerOverflow)?;
+
         let changelog_entry = ChangelogEntry::new(node, changelog_path, leaf_index);
-        if self.changelog_capacity > 0 {
-            self.inc_current_changelog_index()?;
-            self.changelog.push(changelog_entry.clone())?;
-        }
+        self.inc_current_changelog_index()?;
+        self.changelog.push(changelog_entry.clone())?;
```

Comment:
Double checking something that looks suspicious.
This is a bounded vec operation and the changelog_entry is only on the stack, so no vector is used here right?

Reply:
`changelog_entry` is being initially created on the stack and then this line (`self.changelog.push`) creates a copy, which is then stored in the account data (`self.changelog` is a `BoundedVec` backed by the account data).

So yes, no new heap memory is being allocated here.